src/Apropos/HasPermutationGenerator/Morphism.hs

```diff
@@ -63,26 +63,22 @@ seqMorphism a b =
     , morphism = morphism a >=> morphism b
     }
 
-wrapMorphismWithContractCheck :: forall p m. (HasLogicalModel p m, Show m) => Morphism p m -> Morphism p m
-wrapMorphismWithContractCheck mo = mo {morphism = wrap}
+addPropCheck :: forall p m. (HasLogicalModel p m, Show m) => (Set p, Set p) -> Morphism p m -> Morphism p m
+addPropCheck (inps, outps) mo = mo {morphism = wrap}
   where
     wrap :: m -> Gen m
     wrap m = do
```

Comment:
The label has been removed here. We should add
```
label $ name mo
```
inside `wrap`

Reply:
Sorry about that. Fixed now :+1: